### PR TITLE
Add shared source

### LINKED ISSUE #54: Document: feat: enable shared source in session variable by default, and add cluster-level config to disable
This issue tracks the documentation update needed for the merged PR #18749.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/18749
Source PR Merged At: 2024-10-21T02:34:30Z

If it is a major improvement that deserves a new page or a new section in the documentation, please check if we should label it as an experiment feature.

### PR DESCRIPTION
**Pull request by WanYixian**
## Description

- Turn the [draft](https://www.notion.so/risingwave-labs/User-Doc-for-Shared-Source-10ef0d69cb768078b2a5e05bfa5d4807) into technical doc
- Special thanks to Nika for providing the images

## Related Code PR
https://github.com/risingwavelabs/risingwave/pull/18749

## Related Doc PR
Resolve https://github.com/risingwavelabs/risingwave-docs-new/issues/54

## Preview
https://risingwavelabs-wyx-resolve_54.mintlify.app/sql/commands/sql-create-source#shared-source

### CODE REVIEW

**Comment by WanYixian** [comment]
Please help take a look, and let me know if any comments, thanks! @xxchan

**Review by xxchan — APPROVED** [review]
LGTM, thanks!

File: sql/commands/sql-create-source.mdx [hunk 1]
```diff
@@ -88,6 +88,86 @@ If Kafka is part of your technical stack, you can also use the Kafka connector i
...
+ALTER SYSTEM SET streaming_use_shared_source=[true|false];
+```
+
+To completely disable it at the cluster level, go to [`risingwave.toml`](https://github.com/risingwavelabs/risingwave/blob/main/src/config/example.toml#L146) configuration file, and set the `stream_enable_shared_source` to `false`.
+
+### Compared with non-shared source
+
+With non-shared sources, when using the `CREATE SOURCE` statement: 
+- No streaming jobs would be instantiated. A source is just a set of metadata stored in the catalog. 
+- Only when a materialized view references the source, a `SourceExecutor` will be created to start the process of data ingestion.
```
Comment (fuyufjh) [inline sql/commands/sql-create-source.mdx]:
```suggestion
- Only when a materialized view or sink references the source, a `SourceExecutor` will be created to start the process of data ingestion.
```

File: sql/commands/sql-create-source.mdx [hunk 2]
```diff
@@ -88,6 +88,86 @@ If Kafka is part of your technical stack, you can also use the Kafka connector i
...
+Tables offer other features that enhance their utility in data ingestion workflows. See [Table with connectors](/ingestion/overview#table-with-connectors).
+
+<Frame>
+  <img src="/images/table-with-connectors.png"/>
+</Frame>
+
+<Note>
+**LIMITATION**
+
+Currently, shared source is only applicable to Kafka sources. Other sources are unaffected.
```
Comment (fuyufjh) [inline sql/commands/sql-create-source.mdx]:
```suggestion
Currently, shared source is only applicable to Kafka sources. Other sources are unaffected. We plan to gradually upgrade other sources to the be shared as well in the future.
```

**Review by fuyufjh — APPROVED** [review]
LGTM!

**Pull request merged**